examples/02_sphere/main.rs

```diff
@@ -1,45 +1,64 @@
-        use amethyst::renderer::{Layer, Camera, Light};
-        use cgmath::Vector3;
-
-        let (w, h) = ctx.renderer.get_dimensions().unwrap();
-        let proj = Camera::perspective(60.0, w as f32 / h as f32, 1.0, 100.0);
-        let eye = [0.0, 5.0, 0.0];
-        let target = [0.0, 0.0, 0.0];
-        let up = [0.0, 0.0, 1.0];
-        let view = Camera::look_at(eye, target, up);
-        let camera = Camera::new(proj, view);
-
-        ctx.renderer.add_scene("main");
-        ctx.renderer.add_camera(camera, "main");
-
-        ctx.asset_manager.register_asset::<Mesh>();
-        ctx.asset_manager.register_asset::<Texture>();
-        ctx.asset_manager.create_constant_texture("dark_blue", [0.0, 0.0, 0.01, 1.0]);
-        ctx.asset_manager.create_constant_texture("green", [0.0, 1.0, 0.0, 1.0]);
-        ctx.asset_manager.gen_sphere("sphere", 32, 32);
-
-        let translation = Vector3::new(0.0, 0.0, 0.0);
-        let transform: [[f32; 4]; 4] = cgmath::Matrix4::from_translation(translation).into();
-        let fragment = ctx.asset_manager.get_fragment("sphere", "dark_blue", "green", transform).unwrap();
-
-        ctx.renderer.add_fragment("main", fragment);
-
+        use amethyst::renderer::{Layer, Light};
+        use amethyst::world_resources::camera::{Projection, Camera};
+        use amethyst::world_resources::ScreenDimensions;
+        use amethyst::components::transform::LocalTransform;
+        use amethyst::components::rendering::{Texture, Mesh, Renderable};
+        let clear_layer =
+            Layer::new("main",
+                        vec![
+                            Clear::new([0.0, 0.0, 0.0, 1.0]),
+                            DrawShaded::new("main", "main"),
+                        ]);
+        pipeline.layers = vec![clear_layer];
+        {
+            let dimensions = world.read_resource::<ScreenDimensions>();
+            let mut camera = world.write_resource::<Camera>();
+            camera.projection = Projection::Perspective {
+                fov: 90.0,
+                aspect_ratio: dimensions.aspect_ratio,
+                near: 0.1,
+                far: 100.0,
+            };
+            camera.eye = [5.0, 0.0, 0.0];
+            camera.target = [0.0, 0.0, 0.0];
+        }
+        let sphere_vertices = gen_sphere(32, 32);
+        let sphere_mesh = asset_manager.load_asset_from_data::<Mesh, Vec<VertexPosNormal>>(sphere_vertices).unwrap();
+        let dark_blue = asset_manager.load_asset_from_data::<Texture, [f32; 4]>([0.0, 0.0, 0.01, 1.0]).unwrap();
+        let green = asset_manager.load_asset_from_data::<Texture, [f32; 4]>([0.0, 1.0, 0.0, 1.0]).unwrap();
+        let sphere = Renderable {
+            ka: dark_blue,
+            kd: green,
+            mesh: sphere_mesh,
+        };
+        let transform = LocalTransform::default();
+        world.create_now()
+            .with::<Renderable>(sphere)
+            .with::<LocalTransform>(transform)
```

Comment:
I think this needs to also add a `.with::<Transform>(Transform::default())` since the `LocalTransform` only works by putting the calculated position inside of the `Transform` component.